### DISCUSSION TITLE
Ideas on using encode_metadata option in output.hls

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#2023] [General]
I have an icecast hls relay with ```input.http(icecast_source)``` as a source.

```on_metadata``` shows title metadata.

I output with ```output.hls``` with ```encode_metadata``` set to true, but get no metadata in the resulting stream.

I see in the output hls ml file it inserts metadata if m exists so don't see why .

Any clues, am I doing something wrong. 

Thank you.

**Comment by Russsgithub** [comment]
I notice there is an open issue about this.

**Reply by toots** [reply]
Unfortunately, this option does nothing with the `%ffmpeg` encoder. I'll be removing it for `2.0.1`. There are several proposed standards for metadata with HLS but each time I look into it I am unable to find a reference implementation. If you can point me to the right info, I'd be happy to look at it again.

**Comment by Russsgithub** [comment]
Hi Romain,

I'll have a look into it and see what I can find.

Thanks.

On 18 Oct 2021 14:45, Romain Beauxis ***@***.***> wrote:

Unfortunately, this option does nothing with the %ffmpeg encoder. I'll be removing it for 2.0.1. There are several proposed standards for metadata with HLS but each time I look into it I am unable to find a reference implementation. If you can point me to the right info, I'd be happy to look at it again.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/savonet/liquidsoap/discussions/2023#discussioncomment-1495760>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AB627QPMQWQXDBITYYKEGHLUHQQGRANCNFSM5GFBUX3A>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by Russsgithub** [comment]
@toots 

Hi Romain,

I've had a dig around, and found lots of confusing info. 

The most useful to me seemed to be this [script](https://github.com/lebougui/hls-id3tags) , hls-id3tags .

Which uses this [script](https://github.com/dusterio/hlsinjector) , hlsinjector.

Both projects have multiple stars on github which makes me think they work. I've not had a chance to test them, but thought you may gleen some information from them.

There is also ```-metadata title="foo"``` which seems to need ```-map-metadata``` , defaults to id3v2.4 (apparently windows doesn't like this) but can be set to 2.3 with ``` -id3v2_version 3 ``` .
I've seen this work (with ffprobe - it shows it the metadata field) but don't think it is the way to go.

Timed metadata seems to have to do with #EXT-X-DATERANGE and SCTE-35. 

[Here](https://github.com/lexxwork/ffmpeg-scte-35) is a version of ffmpeg with scte-35 implementation, which may offer more info ... maybe.

I guess you may have seen these already , but if not I hope some of it helps.

Russ
